src/probnum/linops/_linear_operator.py

```diff
@@ -343,190 +528,361 @@ def __rmul__(self, other: BinaryOperandType) -> "LinearOperator":
-    ):
-        self._scalar = probnum.utils.as_numpy_scalar(scalar, dtype=dtype)
+            return np.apply_along_axis(rmatvec, -1, x)
 
-        super().__init__(shape=probnum.utils.as_shape(shape), dtype=self._scalar.dtype)
+        return _rmatmul
+
+    @classmethod
+    def broadcast_rmatmat(
+        cls, rmatmat: Callable[[np.ndarray], np.ndarray]
+    ) -> Callable[[np.ndarray], np.ndarray]:
+        def _rmatmul(x: np.ndarray) -> np.ndarray:
+            if x.ndim == 2:
+                return rmatmat(x)
+
+            return _apply_to_matrix_stack(rmatmat, x)
+
+        return _rmatmul
 
     @property
-    def scalar(self):
-        return self._scalar
+    def _inexact_dtype(self) -> np.dtype:
+        if np.issubdtype(self.dtype, np.inexact):
+            return self.dtype
+        else:
+            return np.double
 
-    def _matvec(self, x: np.ndarray) -> np.ndarray:
-        return self._scalar * x
 
-    def _rmatvec(self, x: np.ndarray) -> np.ndarray:
-        return x.T * self._scalar
+def _apply_to_matrix_stack(
+    mat_fn: Callable[[np.ndarray], np.ndarray], x: np.ndarray
+) -> np.ndarray:
+    idcs = np.ndindex(x.shape[:-2])
 
-    def _matmat(self, X: np.ndarray) -> np.ndarray:
-        return self._scalar * X
+    # Shape and dtype inference
+    idx0 = next(idcs)
+    y0 = mat_fn(x[idx0])
 
-    def todense(self):
-        return np.eye(self.shape[0]) * self._scalar
+    # Result buffer
+    y = np.empty(x.shape[:-2] + y0.shape, dtype=y0.dtype)
 
-    def inv(self) -> LinearOperator:
-        return ScalarMult(shape=self.shape, scalar=1 / self._scalar)
+    # Fill buffer
+    y[idx0] = y0
 
-    # Properties
-    def rank(self) -> int:
-        return min(self.shape)
+    for idx in idcs:
+        y[idx] = mat_fn(x[idx])
 
-    def eigvals(self):
-        return np.ones(self.shape[0]) * self._scalar
+    return y
 
-    def det(self):
-        return self._scalar ** self.shape[0]
 
-    def logabsdet(self):
-        return np.log(np.abs(self._scalar))
+class TransposedLinearOperator(LinearOperator):
+    """Transposition of a linear operator."""
 
-    def trace(self):
-        return self._scalar * self.shape[0]
+    def __init__(
+        self,
+        linop: LinearOperator,
+        matmul: Optional[Callable[[np.ndarray], np.ndarray]] = None,
+    ):
+        self._linop = linop
 
+        if matmul is None:
+            matmul = lambda x: self.todense(cache=True) @ x
 
-class Identity(ScalarMult):
-    """The identity operator.
+        super().__init__(
+            shape=(self._linop.shape[1], self._linop.shape[0]),
+            dtype=self._linop.dtype,
+            matmul=matmul,
+            rmatmul=lambda x: self._linop(x, axis=-1),
+            todense=lambda: self._linop.todense(cache=False).T.copy(order="C"),
+            transpose=lambda: self._linop,
+            adjoint=self._linop.conj,
+            conjugate=self._linop.adjoint,
+            inverse=None,  # lambda: self._linop.inv().T,
```

Comment:
Is this commented out on purpose?

Reply:
Yes, this will in the future be set depending on whether the linear operator has a custom inverse and if that inverse has an efficient transpose, but I'm not quite there yet